projects/kit/src/lib/utils/date/validate-date-string.ts

```diff
@@ -22,7 +25,11 @@ export function validateDateString({
     const validatedDateSegments: Partial<MaskitoDateSegments> = {};
 
     for (const [segmentName, segmentValue] of dateSegments) {
-        const validatedDate = toDateString(validatedDateSegments, dateModeTemplate);
+        const validatedDate = toDateString(
+            validatedDateSegments,
+            dateModeTemplate,
+            dateTimeSeparator,
```

Comment:
It it really required here ?

I've understood everything correctly,  `dateTimeSeparator` is required only if `timeMode` is passed too.

Reply:
Probably, to avoid the same problem in future we need to add overload

```ts
export function toDateString(
    segments: Partial<MaskitoDateSegments & Partial<MaskitoTimeSegments>>,
    dateMode: string,
): string;

export function toDateString(
    segments: Partial<MaskitoDateSegments & Partial<MaskitoTimeSegments>>,
    dateMode: string,
    timeMode: string,
    dateTimeSeparator: string,
): string;
```

Reply:
I didn't add overload. I just wrapped `dateMode` `timeMode` `dateTimeSeparator` into an object and made `timeMode` and `dateTimeSeparator` optional values.